backend/src/main/kotlin/com/infowings/catalog/data/aspect/AspectService.kt

```diff
@@ -184,6 +186,55 @@ class DefaultAspectService(
         }
     }
 
+    /**
+     * Removes [AspectPropertyVertex] if it is not linked.
+     * Marks [AspectPropertyVertex] as removed if it is linked and [force] is true.
+     * @param aspectPropertyId [AspectPropertyVertex.id] of corresponding [AspectPropertyVertex]
+     * @param force Flag that allows to mark [AspectPropertyVertex] as removed by setting its [AspectPropertyVertex.deleted] flag to true if
+     * [AspectPropertyVertex] is linked
+     */
+    override fun removeProperty(aspectPropertyId: String, username: String, force: Boolean): AspectPropertyDeleteResponse {
+        val removePropertyResult = transaction(db) {
+            val userVertex = userService.findUserVertexByUsername(username)
+            val historyContext = HistoryContext(userVertex)
+
+            val aspectPropertyVertex = aspectDaoService.findProperty(aspectPropertyId) ?: throw AspectPropertyDoesNotExist(aspectPropertyId)
+            val aspectPropertyIsLinked = aspectPropertyVertex.isLinkedBy()
+
+            val parentAspectVertex = aspectPropertyVertex.getVertices(ODirection.IN, ASPECT_ASPECT_PROPERTY_EDGE).single().toAspectVertex()
+            val childAspectVertex = aspectPropertyVertex.associatedAspect
+
+            when {
+                aspectPropertyIsLinked && force -> {
+                    val propertySoftDeleteFact = aspectPropertyVertex.toSoftDeleteFact(historyContext)
+                    historyService.storeFact(propertySoftDeleteFact)
+                    aspectDaoService.fakeRemove(aspectPropertyVertex)
+                }
+                aspectPropertyIsLinked && !force -> throw AspectPropertyIsLinkedByValue(aspectPropertyVertex.id)
+                else -> {
+                    val aspectSnapshot = parentAspectVertex.currentSnapshot()
+
+                    val propertyDeleteFact = aspectPropertyVertex.toDeleteFact(historyContext)
+                    historyService.storeFact(propertyDeleteFact)
+                    aspectDaoService.remove(aspectPropertyVertex)
+
+                    historyService.storeFact(parentAspectVertex.toUpdateFact(historyContext, aspectSnapshot))
+                }
+            }
+            AspectPropertyDeleteResult(aspectPropertyVertex, parentAspectVertex, childAspectVertex)
+        }
+
+        return removePropertyResult.toResponse()
+    }
+
+    private fun AspectPropertyDeleteResult.toResponse() = AspectPropertyDeleteResponse(
+        this.aspectPropertyVertex.id,
+        PropertyCardinality.valueOf(this.aspectPropertyVertex.cardinality),
+        this.aspectPropertyVertex.name,
+        Reference(this.parentAspectVertex.id, this.parentAspectVertex.version),
+        Reference(this.childAspectVertex.id, this.childAspectVertex.version)
+    )
+
```

Comment:
I guess that best place for toResponse method is exactly inside AspectPropertyDeleteResult class definition.
Just because this option should be considered by default. 
I see just 2 possible reasons to move it outside of class definition:
* it is third-party library class
* there is some context necessary for method logic and this context is absent inside class definition
None of these treason is applicable here.